### PR TITLE
fix(postcss-merge-rules): check for ms vendor

### LINKED ISSUE #419: mergeRules incorrectly merges placeholder selectors
For example, consider this CSS file:

```css
input:-ms-input-placeholder {
  font-size: 14px;
}
input::-ms-input-placeholder {
  font-size: 14px;
}
```

If you run cssnano against this file with the default preset, you get:

```css
input:-ms-input-placeholder,input::-ms-input-placeholder{font-size:14px}
```

Unfortunately, this won't work with Internet Explorer. These rules cannot be safely merged. When I disable the `mergeRules` optimization, this issue is resolved.

This issue may be related to https://github.com/ben-eb/cssnano/issues/297

### PR DESCRIPTION
**Pull request by evless**
fixes #419

### CODE REVIEW

**Comment by ai** [comment]
Maybe we should check not just prefix but whole selector? To be sure that it is placeholder selector?

**Comment by evilebottnawi** [comment]
Yep, looks like better use `-ms-input-placeholder` whole selector and add comment to future developers:
```
Internet Explorer use :-ms-input-placeholder.
Microsoft Edge use ::-ms-input-placeholder.
```

File: packages/postcss-merge-rules/src/index.js [hunk 1]
```diff
@@ -13,15 +13,17 @@ function intersect (a, b, not) {
     });
 }
 
-const checkMS = (a, b) => !!(~a.indexOf('ms') && ~b.indexOf('ms'));
+
+// Internet Explorer use :-ms-input-placeholder.
+// Microsoft Edge use ::-ms-input-placeholder.
+const findMsInputPlaceholder = selector => ~selector.indexOf('-ms-input-placeholder');
```
Comment (evilebottnawi) [inline packages/postcss-merge-rules/src/index.js]: Better use `regex` with `i` flag, because css is case insensitivity language

Comment (evless) [reply packages/postcss-merge-rules/src/index.js]: Oh, well, I did not think about it

**Comment by evilebottnawi** [comment]
@evless Thanks for work!

**Comment by evless** [comment]
@evilebottnawi I apologize, but can I make a commit with the addition of a list of contributors? :)

**Comment by evilebottnawi** [comment]
@evless yep, feel free to send PR :+1:

**Pull request merged**